modules/candidate_list/test/candidate_listTest.php

```diff
@@ -77,56 +90,212 @@ function testCandidateListPageLoads()
-
         $dobOptions = $this->webDriver->findElement(WebDriverBy::Name("dob"));
         $this->assertEquals("input", $dobOptions->getTagName());
         // Not currently done
         //$this->assertEquals("date",$dobOptions->getAttribute("type"));
-
         $genderOptions = $this->webDriver->findElement(WebDriverBy::Name("gender"));
         $this->assertEquals("select", $genderOptions->getTagName());
-
         $numVisits = $this->webDriver->findElement(WebDriverBy::Name("Visit_Count"));
         $this->assertEquals("input", $dobOptions->getTagName());
         // Not currently done in Loris.
         //$this->assertEquals("number",$dobOptions->getAttribute("type"));
         //$this->assertEquals("0",$dobOptions->getAttribute("min"));
-
         $edcOptions = $this->webDriver->findElement(WebDriverBy::Name("edc"));
         $this->assertEquals("input", $edcOptions->getTagName());
         // Not currently done
         //$this->assertEquals("date",$edcOptions->getAttribute("type"));
-
         $latestVisitOptions = $this->webDriver->findElement(
             WebDriverBy::Name("Latest_Visit_Status")
         );
         $this->assertEquals("select", $latestVisitOptions->getTagName());
-
         $feedbackOptions = $this->webDriver->findElement(
             WebDriverBy::Name("Feedback")
         );
         $this->assertEquals("select", $feedbackOptions->getTagName());
     }
+    /**
+     * Performs various searches by PSCID (and PSCID only).
+     *
+     * @return void.
+     */
+    function testFilterByPscid()
+    {
+        $this->safeGet($this->url . "/candidate_list/");
+        // Enter something that does not even make sense in the PSCID field
+        // Verify that no candidates are returned
+        $this->_assertSearchBy(
+            array('PSCID' => 'PSCID that does not exist'),
+            null
+        );
+        // Search using a PSCID that does not exist
+        // Verify that no candidates are returned
+        $this->_assertSearchBy(
+            array('PSCID' => 'TST0003'),
+            null
+        );
+        // Search using PSCID TST0001
+        // Verify that only one candidate is returned: TST0001
+        $this->_assertSearchBy(
+            array('PSCID' => 'TST0001'),
+            array(self::$_TST0001)
+        );
+        // Search for candidate with PSCID tst0001
+        // Verify that candidate TST0001 is returned (checks that searches
+        // are case-insensitive)
+        $this->_assertSearchBy(
+            array('PSCID' => 'tst0001'),
+            array(self::$_TST0001)
+        );
+        // Search for PSCID that contains string t0
+        // Verify that candidate TST0001 is returned
+        $this->_assertSearchBy(
+            array('PSCID' => 't0'),
+            array(self::$_TST0001)
+        );
+    }
+    /**
+     * Performs various searches by DCCID (and DCCID only).
+     *
+     * @return void.
+     */
+    function testFilterByDccId()
+    {
```

Comment:
// Stop here and mark this test as incomplete.
        $this->markTestIncomplete(
          'This test is using a React component'
        );